src/map/entities/battleentity.cpp

```diff
@@ -206,6 +206,19 @@ uint8 CBattleEntity::GetHPP() const
     return hpp;
 }
 
+uint8 CBattleEntity::GetHPPNoPercentOrConvert()
+{
+    TracyZoneScoped;
+    // recalculate max HP using only raw HP mods; no convert or HPP, regardless of source, apply
+    int32 modHP = std::max(1, health.maxhp + getMod(Mod::HP));
+
+    uint8 hpp = (uint8)floor(((float)health.hp / (float)modHP) * 100);
```

Comment:
I think you're implicitly casting to an int too early here?

Reply:
Would you be able to explain more? I wanted to mirror `CBattleEntity::GetHPP`'s logic here for HPP calculation.

Reply:
Ah, I thought that * 100 would cast to an int here, my misunderstanding!